tests/aws/services/apigateway/test_apigateway_lambda_cfn.py

```diff
@@ -1,80 +1,304 @@
+
+        region = aws_client.events.meta.region_name
+        invocation_endpoint = f"https://{api_id}.execute-api.{region}.amazonaws.com/prod/test"
+        connection_name = f"c-{short_uid()}"
+        auth_type = auth["type"]
+        key = auth["key"]
+        parameters = auth["parameters"].copy()
+
+        if auth_type == "OAUTH_CLIENT_CREDENTIALS":
+            parameters["AuthorizationEndpoint"] = oauth_token_url
+
+        connection_resp = aws_client.events.create_connection(
+            Name=connection_name,
+            AuthorizationType=auth_type,
+            AuthParameters={
+                key: parameters,
+                "InvocationHttpParameters": {
+                    "BodyParameters": [
+                        {"Key": "connection_body_param", "Value": "value", "IsValueSecret": False},
+                        {"Key": "oauthbody", "Value": "value1", "IsValueSecret": False},
+                    ],
+                    "HeaderParameters": [
+                        {
+                            "Key": "connection-header-param",
+                            "Value": "value",
+                            "IsValueSecret": False,
+                        },
+                        {"Key": "overwritten-header", "Value": "original", "IsValueSecret": False},
+                        {"Key": "oauthheader", "Value": "value2", "IsValueSecret": False},
+                    ],
+                    "QueryStringParameters": [
+                        {"Key": "connection_query_param", "Value": "value", "IsValueSecret": False},
+                        {"Key": "overwritten_query", "Value": "original", "IsValueSecret": False},
+                        {"Key": "oauthquery", "Value": "value3", "IsValueSecret": False},
+                    ],
+                },
+            },
+        )
+
+        if auth_type == "OAUTH_CLIENT_CREDENTIALS":
+
+            def _wait_for_connection_auth():
+                conn_desc = aws_client.events.describe_connection(Name=connection_name)
+                state = conn_desc.get("ConnectionState")
+                if state == "AUTHORIZED":
+                    return conn_desc
+                raise AssertionError(f"Connection not yet authorized. Current state: {state}")
+
+            retry(_wait_for_connection_auth, retries=20, sleep=5)
+
+        dest_name = f"d-{short_uid()}"
+        dest_resp = aws_client.events.create_api_destination(
+            Name=dest_name,
+            ConnectionArn=connection_resp["ConnectionArn"],
+            InvocationEndpoint=invocation_endpoint,
+            HttpMethod="POST",
+        )
+        api_destination_arn = dest_resp["ApiDestinationArn"]
+
+        time.sleep(10)
+
+        rule_name = f"r-{short_uid()}"
+        target_id = f"target-{short_uid()}"
+        pattern = json.dumps({"source": ["source-123"], "detail-type": ["type-123"]})
+        aws_client.events.put_rule(Name=rule_name, EventPattern=pattern)
+        time.sleep(10)
```

Comment:
question: do we need need this sleep in LocalStack, or only in AWS? this could be conditional, we tend to avoid sleeping in test as much as we can.